src/Plugin/Validation/Constraint/GlobalMessageConstraintValidator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Drupal\stanford_profile_helper\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the UniqueInteger constraint.
+ */
+class GlobalMessageConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+
+    $is_valid = FALSE;
+
+    $resource_fields = [
+      'su_global_msg_label',
+      'su_global_msg_header',
+      'su_global_msg_message',
+      'su_global_msg_link',
+    ];
+
+    if ($value->getEntity()->hasField('su_global_msg_enabled') &&
+    $value->getEntity()->get('su_global_msg_enabled')->getValue()[0]['value']) {
```

Comment:
`getString()` should suffice here. 
```suggestion
    if (
      $value->getEntity()->hasField('su_global_msg_enabled') &&
      $value->getEntity()->get('su_global_msg_enabled')->getString()
    ) {
```

Reply:
Ok, I'll change the unit test as well.

Reply:
actually, this is going to break my test; the getString() method gets mocked to return string values for the fields, but for just that one field, I need it to always come back true.

Reply:
awesome. Also I find that in a multiple line `if` statement, it's easier to read it like i suggested here. The indentation of the conditions make its more obvious.

Reply:
Without running the tests myself, you should be able to just change [line 79](https://github.com/SU-SWS/stanford_profile_helper/pull/88/files#diff-4e7532b41f227398c42143faa8ff243e9ca4afd4e8f913d3e424c827fb728f3bR79) to be a string instead of an array.